app/src/main/java/com/google/android/sambadocumentsprovider/document/DocumentMetadata.java

```diff
@@ -74,6 +75,19 @@ public boolean isFileShare() {
     return mEntry.getType() == DirectoryEntry.FILE_SHARE;
   }
 
+  public int getIconResourceId() {
+    switch (mEntry.getType()) {
+      case DirectoryEntry.BROWSING_ROOT:
+        return R.drawable.ic_cloud;
+      case DirectoryEntry.SERVER:
+        return R.drawable.ic_server;
+      case DirectoryEntry.FILE_SHARE:
+        return R.drawable.ic_folder_shared;
+      default:
+        return R.drawable.ic_folder;
```

Comment:
I don't think you need ic_folder. Return -1 to use the default one.

Or better just don't return that column.